### PR TITLE
Add user for photon and delete some needless files

### PR DESCRIPTION
**Pull request by zhengxiexie**
Add user nsx-ncp in Dockerfile, so nsx-operator could run as user nsx-ncp, this is required by security problem.

### CODE REVIEW

File: build/image/photon/Dockerfile [hunk 1]
```diff
@@ -9,6 +9,9 @@ RUN go build -o manager cmd/main.go
 
 FROM photon
 
+RUN tdnf -y install shadow && \
+    useradd -s /bin/bash nsx-operator
+
 COPY --from=golang-build /source/manager /usr/local/bin/
 
-ENTRYPOINT [ "/usr/local/bin/manager"]
+ENTRYPOINT ["su", "nsx-ncp", "-m", "-c", "/usr/local/bin/manager", "-nsxconfig", "/etc/nsx-ujo/ncp.ini", "-health-probe-bind-address", ":8384"]
```
Comment (dantingl) [inline build/image/photon/Dockerfile:17]: `nsx-ncp` -> nsx-operator
`-health-probe-bind-address`: Do we need this as the default port is 8384 in code.

Comment (dantingl) [inline build/image/photon/Dockerfile:17]: `/etc/nsx-ujo/ncp.ini`: Sorry, this does not look good to me, that we add ncp config in nsx-operator repo, so looks like we still need command in NCP yaml definition.
@heypnus any comments?

Comment (zhengxiexie) [reply build/image/photon/Dockerfile:17]: ok

Comment (zhengxiexie) [reply build/image/photon/Dockerfile:17]: ok

File: build/image/photon/Dockerfile [hunk 2]
```diff
@@ -9,6 +9,9 @@ RUN go build -o manager cmd/main.go
 
 FROM photon
 
+RUN tdnf -y install shadow && \
+    useradd -s /bin/bash nsx-operator
+
 COPY --from=golang-build /source/manager /usr/local/bin/
 
-ENTRYPOINT [ "/usr/local/bin/manager"]
+ENTRYPOINT [ "/usr/local/bin/manager"]
```
Comment (dantingl) [inline build/image/photon/Dockerfile:17]: Here I think we need to switch to nsx-operator user

File: build/image/photon/Dockerfile [hunk 3]
```diff
@@ -9,6 +9,9 @@ RUN go build -o manager cmd/main.go
 
 FROM photon
 
+RUN tdnf -y install shadow && \
+    useradd -s /bin/bash nsx-operator
+
 COPY --from=golang-build /source/manager /usr/local/bin/
```
Comment (dantingl) [inline build/image/photon/Dockerfile:16]: Use `USER nsx-operator` to switch user

**Review by dantingl — APPROVED** [review]

**Pull request merged**